RealmSwift/List.swift

```diff
@@ -0,0 +1,310 @@
+
+    :returns: The object at the given `index`.
+    */
+    public subscript(index: Int) -> T {
+        get {
+            throwForNegativeIndex(index)
+            return _rlmArray[UInt(index)] as T
+        }
+        set {
+            throwForNegativeIndex(index)
+            return _rlmArray[UInt(index)] = newValue
+        }
+    }
+
+    /// Returns the first object in the list, or `nil` if empty.
+    public var first: T? { return _rlmArray.firstObject() as T? }
+
+    /// Returns the last object in the list, or `nil` if empty.
+    public var last: T? { return _rlmArray.lastObject() as T? }
+
+    // MARK: Filtering
+
+    /**
+    Returns `Results` containing list elements that match the given predicate.
+
+    :param: predicateFormat The predicate format string which can accept variable arguments.
+
+    :returns: `Results` containing list elements that match the given predicate.
+    */
+    public func filter(predicateFormat: String, _ args: CVarArgType...) -> Results<T> {
+        return Results<T>(_rlmArray.objectsWithPredicate(NSPredicate(format: predicateFormat, arguments: getVaList(args))))
+    }
+
+    /**
+    Returns `Results` containing list elements that match the given predicate.
+
+    :param: predicate The predicate to filter the objects.
+
+    :returns: `Results` containing list elements that match the given predicate.
+    */
+    public func filter(predicate: NSPredicate) -> Results<T> {
+        return Results<T>(_rlmArray.objectsWithPredicate(predicate))
+    }
+
+    // MARK: Sorting
+
+    /**
+    Returns `Results` containing list elements sorted by the given property.
+
+    :param: property  The property name to sort by.
+    :param: ascending The direction to sort by.
+
+    :returns: `Results` containing list elements sorted by the given property.
+    */
+    public func sorted(property: String, ascending: Bool = true) -> Results<T> {
+        return sorted([SortDescriptor(property: property, ascending: ascending)])
+    }
+
+    /**
+    Returns `Results` with elements sorted by the given sort descriptors.
+
+    :param: sortDescriptors `SortDescriptor`s to sort by.
+
+    :returns: `Results` with elements sorted by the given sort descriptors.
+    */
+    public func sorted<S: SequenceType where S.Generator.Element == SortDescriptor>(sortDescriptors: S) -> Results<T> {
+        return Results<T>(_rlmArray.sortedResultsUsingDescriptors(map(sortDescriptors) { $0.rlmSortDescriptorValue }))
+    }
+
+    // MARK: Sequence Support
+
+    /// Returns a `GeneratorOf<T>` that yields successive elements in the list.
+    public func generate() -> GeneratorOf<T> {
+        var i: UInt = 0
+        return GeneratorOf<T>() {
+            if (i >= self._rlmArray.count) {
+                return .None
+            } else {
+                return self._rlmArray[i++] as? T
+            }
+        }
+    }
+
+    // MARK: Mutation
+
+    /**
+    Appends the given object to the end of the list.
+
```

Comment:
Adds the object to the List's Realm if the object is standalone and the List is persisted in a Realm, and copies the object to the List's Realm if they are persisted in different Realms.